scripts/localnet/roles/bittensor_wallet/tasks/main.yml

```diff
@@ -0,0 +1,244 @@
+  when: not net2_exists
+  tags: bittensor_wallet
+
+- name: Check if wallet has funds already
+  shell: btcli wallet overview --wallet.name "{{ cold_wallet_name }}" --subtensor.network local
+  register: wallet_overview
+  when: not net2_exists
+  tags: bittensor_wallet
+  no_log: false
+  args:
+    executable: /bin/bash
+
+- name: Debug wallet overview
+  debug:
+    msg: "{{ wallet_overview.stdout_lines }}"
+  when: not net2_exists
+  tags: bittensor_wallet
+
+- name: Fund cold wallet with environment variable password
+  shell: >
+    {{ wallet_env_var }}=$(python -c "from bittensor_wallet import Wallet; print(Wallet('{{ cold_wallet_name }}').coldkey_file.save_password_to_env('{{ bittensor_wallet_password }}'))")
+    btcli wallet faucet --wallet.name "{{ cold_wallet_name }}" --subtensor.network local -y --max-successes 3
+  register: direct_faucet_result
+  when: not net2_exists
+  tags: bittensor_wallet
+  no_log: false
+  args:
+    executable: /bin/bash
+
+- name: Debug direct faucet result
+  debug:
+    msg: "{{ direct_faucet_result.stdout_lines | default([]) + direct_faucet_result.stderr_lines | default([]) }}"
+  when: not net2_exists
+  tags: bittensor_wallet
+
+- name: Check wallet balance after faucet
+  shell: btcli wallet overview --wallet.name "{{ cold_wallet_name }}" --subtensor.network local
+  register: wallet_after_faucet
+  when: not net2_exists
+  tags: bittensor_wallet
+  no_log: false
+  args:
+    executable: /bin/bash
+
+- name: Debug wallet after faucet
+  debug:
+    msg: "{{ wallet_after_faucet.stdout_lines }}"
+  when: not net2_exists
+  tags: bittensor_wallet
+
+- name: Check local network status
+  shell: btcli subnet list --subtensor.network local
+  register: subnet_list
+  when: not net2_exists
+  tags: bittensor_wallet
+  no_log: false
+  args:
+    executable: /bin/bash
+
+- name: Debug subnet list
+  debug:
+    msg: "{{ subnet_list.stdout_lines }}"
+  when: not net2_exists
+  tags: bittensor_wallet
+
+- name: Check balance before subnet creation
+  shell: btcli wallet overview --wallet.name "{{ cold_wallet_name }}" --subtensor.network local
+  register: balance_before_subnet
+  when: not net2_exists
+  tags: bittensor_wallet
+  no_log: false
+  args:
+    executable: /bin/bash
+
+- name: Debug balance before subnet creation
+  debug:
+    msg: "{{ balance_before_subnet.stdout_lines }}"
+  when: not net2_exists
+  tags: bittensor_wallet
+
+- name: Subnet creation
+  shell: >
+    export {{ wallet_env_var }}=$(python -c "from bittensor_wallet import Wallet; print(Wallet('{{ cold_wallet_name }}').coldkey_file.save_password_to_env('{{ bittensor_wallet_password }}'))");
+    printf "\n\n\n\n\n\n" | btcli subnet create --wallet.name "{{ cold_wallet_name }}" --wallet.hotkey "{{ owner_hotkey }}" --subtensor.network local --subnet-name "templar_subnet" -y
+  register: direct_subnet_create
+  when: not net2_exists
+  tags: bittensor_wallet
+  no_log: false
+  args:
+    executable: /bin/bash
```

Comment:
Did we not set which subnet to create. Maybe this is not possible but maybe then we should not `netuid` in the vault?

Reply:
from my understanding it's expected to be net id 2. no?

Reply:
Mostly yes. The first subnet we will create in a local test setup should always be subnet 2 I think. But if we are running multiple tests across multiple subnets this may vary. I don't think we need to worry about this so you could ignore my comment.

Reply:
I was mainly wondering why we specified it in some vault settings.